### ISSUE TITLE
engine->flutter roller blocked on customer testing failures

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#137298]
Example:
 - https://github.com/flutter/flutter/pull/137291 
 - https://ci.chromium.org/ui/p/flutter/builders/try/Windows%20customer_testing/65851/overview

It looks like the tests are passing but there is an exception in the code following all the tests passing, this may be a Dart integration problem?

## log
```
| 00:13 +166: All tests passed!
Tests finished in 36.00 seconds per iteration.
Deleting temporary directory...

Processing .\registry\provider.test...
Created temporary directory: C:\b\s\w\ir\x\t\flutter_customer_testing.provider.dce80856
>> git clone https://github.com/rrousselGit/provider.git tests
| Cloning into 'tests'...
>> git -C tests checkout a60a35c49e907fed597dd2af28ed79cc94b7671a
| HEAD is now at a60a35c Fix CI (#810)
Updating code in C:\b\s\w\ir\x\t\flutter_customer_testing.provider.dce80856\tests\....
>> flutter packages get
| Resolving dependencies...
| + _fe_analyzer_shared 61.0.0 (65.0.0 available)
| + analyzer 5.13.0 (6.3.0 available)
| + args 2.4.2
| + async 2.11.0
| + boolean_selector 2.1.1
| + build 2.4.1
| + built_collection 5.1.1
| + built_value 8.6.3
| + characters 1.3.0
| + clock 1.1.1
| + code_builder 4.7.0
| + collection 1.18.0
| + convert 3.1.1
| + coverage 1.7.1
| + crypto 3.0.3
| + dart_style 2.3.2 (2.3.3 available)
| + fake_async 1.3.1
| + file 7.0.0
| + fixnum 1.1.0
| + flutter 0.0.0 from sdk flutter
| + flutter_test 0.0.0 from sdk flutter
| + frontend_server_client 3.2.0
| + glob 2.1.2
| + http_multi_server 3.2.1
| + http_parser 4.0.2
| + intl 0.18.1
| + io 1.0.4
| + js 0.6.7
| + leak_tracker 9.0.11
| + leak_tracker_flutter_testing 1.0.9
| + leak_tracker_testing 1.0.5
| + logging 1.2.0
| + matcher 0.12.16
| + material_color_utilities 0.8.0
| + meta 1.11.0
| + mime 1.0.4
| + mockito 5.4.2
| + nested 1.0.0
| + node_preamble 2.0.2
| + package_config 2.1.0
| + path 1.8.3
| + pool 1.5.1
| + pub_semver 2.1.4
| + shelf 1.4.1
| + shelf_packages_handler 3.0.2
| + shelf_static 1.1.2
| + shelf_web_socket 1.0.4
| + sky_engine 0.0.99 from sdk flutter
| + source_gen 1.4.0
| + source_map_stack_trace 2.1.1
| + source_maps 0.10.12
| + source_span 1.10.0
| + stack_trace 1.11.1
| + stream_channel 2.1.2
| + string_scanner 1.2.0
| + term_glyph 1.2.1
| + test 1.24.9
| + test_api 0.6.1
| + test_core 0.5.9
| + typed_data 1.3.2
| + vector_math 2.1.4
| + vm_service 12.0.0 (13.0.0 available)
| + watcher 1.1.0
| + web 0.3.0
| + web_socket_channel 2.4.0
| + webkit_inspection_protocol 1.2.1
| + yaml 3.1.2
| Changed 67 dependencies!
| 4 packages have newer versions incompatible with dependency constraints.
| Try `flutter pub outdated` for more information.
| Resolving dependencies in .\example...
| Got dependencies in .\example.
>> dart fix --apply
| Computing fixes in tests...
| Error from the analysis server: Exception while getting bulk fixes: type 'NotLibraryButPartResult' is not a subtype of type 'ParsedLibraryResult' in type cast
| #0      BulkFixProcessor._computeChangesToPubspec (package:analysis_server/src/services/correction/bulk_fix_processor.dart:373:13)
| #1      BulkFixProcessor.fixPubspec (package:analysis_server/src/services/correction/bulk_fix_processor.dart:284:7)
| #2      EditBulkFixes.handle (package:analysis_server/src/handler/legacy/edit_bulk_fixes.dart:56:38)
| #3      LegacyAnalysisServer.handleRequest.<anonymous closure>.<anonymous closure> (package:analysis_server/src/legacy_analysis_server.dart:559:25)
| #4      OperationPerformanceImpl.runAsync (package:analyzer/src/util/performance/operation_performance.dart:172:29)
| #5      LegacyAnalysisServer.handleRequest.<anonymous closure> (package:analysis_server/src/legacy_analysis_server.dart:540:29)
| #6      _rootRun (dart:async/zone.dart:1399:13)
| #7      _CustomZone.run (dart:async/zone.dart:1301:19)
| #8      _runZoned (dart:async/zone.dart:1804:10)
| #9      runZonedGuarded (dart:async/zone.dart:1792:12)
| #10     LegacyAnalysisServer.handleRequest (package:analysis_server/src/legacy_analysis_server.dart:536:5)
| #11     LegacyAnalysisServer.handleRequestOrResponse (package:analysis_server/src/legacy_analysis_server.dart:600:7)
| #12     _rootRunUnary (dart:async/zone.dart:1407:47)
| #13     _CustomZone.runUnary (dart:async/zone.dart:1308:19)
| #14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1217:7)
| #15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
| #16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
| #17     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:63:11)
| #18     _EventSinkWrapper.add (dart:async/stream_transformers.dart:13:11)
| #19     new _DebounceRequests.<anonymous closure>.<anonymous closure> (package:analysis_server/src/server/debounce_requests.dart:50:20)
| #20     _rootRun (dart:async/zone.dart:1391:47)
| #21     _CustomZone.run (dart:async/zone.dart:1301:19)
| #22     _CustomZone.runGuarded (dart:async/zone.dart:1209:7)
| #23     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1249:23)
| #24     _rootRun (dart:async/zone.dart:1399:13)
| #25     _CustomZone.run (dart:async/zone.dart:1301:19)
| #26     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:1233:23)
| #27     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:18:15)
| #28     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:398:19)
| #29     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5)
| #30     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
|
|
| #0      BulkFixProcessor._computeChangesToPubspec (package:analysis_server/src/services/correction/bulk_fix_processor.dart:373:13)
| #1      BulkFixProcessor.fixPubspec (package:analysis_server/src/services/correction/bulk_fix_processor.dart:284:7)
| #2      EditBulkFixes.handle (package:analysis_server/src/handler/legacy/edit_bulk_fixes.dart:56:38)
| #3      LegacyAnalysisServer.handleRequest.<anonymous closure>.<anonymous closure> (package:analysis_server/src/legacy_analysis_server.dart:559:25)
| #4      OperationPerformanceImpl.runAsync (package:analyzer/src/util/performance/operation_performance.dart:172:29)
| #5      LegacyAnalysisServer.handleRequest.<anonymous closure> (package:analysis_server/src/legacy_analysis_server.dart:540:29)
| #6      _rootRun (dart:async/zone.dart:1399:13)
| #7      _CustomZone.run (dart:async/zone.dart:1301:19)
| #8      _runZoned (dart:async/zone.dart:1804:10)
| #9      runZonedGuarded (dart:async/zone.dart:1792:12)
| #10     LegacyAnalysisServer.handleRequest (package:analysis_server/src/legacy_analysis_server.dart:536:5)
| #11     LegacyAnalysisServer.handleRequestOrResponse (package:analysis_server/src/legacy_analysis_server.dart:600:7)
| #12     _rootRunUnary (dart:async/zone.dart:1407:47)
| #13     _CustomZone.runUnary (dart:async/zone.dart:1308:19)
| #14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1217:7)
| #15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
| #16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
| #17     _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:63:11)
| #18     _EventSinkWrapper.add (dart:async/stream_transformers.dart:13:11)
| #19     new _DebounceRequests.<anonymous closure>.<anonymous closure> (package:analysis_server/src/server/debounce_requests.dart:50:20)
| #20     _rootRun (dart:async/zone.dart:1391:47)
| #21     _CustomZone.run (dart:async/zone.dart:1301:19)
| #22     _CustomZone.runGuarded (dart:async/zone.dart:1209:7)
| #23     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1249:23)
| #24     _rootRun (dart:async/zone.dart:1399:13)
| #25     _CustomZone.run (dart:async/zone.dart:1301:19)
| #26     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:1233:23)
| #27     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:18:15)
| #28     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:398:19)
| #29     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5)
| #30     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
|
Terminate batch job (Y/N)? 
```

**Comment by jason-simmons** [comment]
`dart fix` in the latest Dart SDK is applying changes to `pubspec.yaml` which add a line that references the package's own name.

For example, running `dart fix` in the flutter_svg package adds a `flutter_svg: any` dependency line to `pubspec.yaml`.

`pub get` then fails with an error:
```
ERR : Error on line 19, column 3 of pubspec.yaml: A package may not list itself as a dependency.
    |    ╷
    | 19 │   flutter_svg: any
```

This was introduced by https://dart.googlesource.com/sdk/+/3c6ceb851ada75197cafd6e839df3e8086539629

**Comment by whesse** [comment]
I think this may be fixed by https://dart-review.googlesource.com/c/sdk/+/332242 which just landed.

**Comment by whesse** [comment]
Oops, sorry, it is clearly due to the pubspec fix added to dart fix, and a revert is already created for that (but not landed):

https://dart-review.googlesource.com/c/sdk/+/332171

**Comment by gaaclarke** [comment]
Here is the commit with the revert: https://dart.googlesource.com/sdk/+/ed1842480c966b93ecf80627178dc2238d6f0e2a

That revert made it into https://dart.googlesource.com/sdk/+/refs/tags/3.3.0-65.0.dev.

Closing this issue although the dart roll is still blocked.  I'll file a new issue representing the latest stoppage.
